autonomy/deploy/generators/docker_compose/base.py

```diff
@@ -170,9 +184,26 @@ def generate(self, image_version: Optional[str] = None) -> "DockerComposeGenerat
                 for i in range(self.service_spec.service.number_of_agents)
             ]
         )
+
+        hardhat_node = ""
```

Comment:
why not default value `None`?

Reply:
It'll will require and extra check later when formatting the final template since we only want to include the hardhat node if the flag is true. We're going to set it to empty string either way.